changelog/6977.improvement.md

```diff
@@ -0,0 +1,11 @@
+[Forms](forms.mdx) no longer reject their execution before a potential custom 
+action for validating / extracting slots was executed. 
+Forms continue to reject in two cases automatically:
+- A slot was requested to be filled, but no slot mapping applied to the latest user
+  message and there was no custom action for potentially extracting other slots.
+- A slot was requested to be filled, but the custom action for validating / extracting
+  slots didn't return any slot event.
```

Comment:
what happens if you want to simply utter "please rephrase" in your custom validation without setting any slot?

Reply:
In that case they'd do that in response to invalid slots, correct? As long as they return `SlotSet(<slot name>, None)` events there won't be an automatic rejection.

Reply:
aaaah, so you'd still need to set requested slot to `None`, I think it is a bit of a hassle from user perspective but could work if we document it well. @ArjaanBuijk what do you think?

Reply:
> so you'd still need to set requested slot to None

No. When the action server retrieves a request to run the validation action **and** this doesn't return any `SlotSet` events at all, then we automatically raise ` ActionExecutionRejected`. If the validation action marks all slots as invalid (by returning `SlotSet(<slot name>, None)` events) then we don't raise a rejection. They can request next slots / do utterances as they desire.

I think this in line with the Rasa Open Source 1.x behavior to be honest. If you use the `FormValidation` action then you'd always return something like `{"slot_name": <value>}`. So not returning any `SlotSet` event in the validation action should really be the exception.

Reply:
sorry, that's what I meant. I didn't finish the sentence that was in my head:
"you'd still need to set requested slot to None if you don't want to reject the form"

Reply:
@wochinge could you please write documentation for this as part of this PR

Reply:
done 👍